### PR TITLE
Support parsing message options of type enum and file level enums.

### PR DESCRIPTION
**Pull request by ninowalker**

### CODE REVIEW

File: parser/protofile.js [hunk 1]
```diff
@@ -86,14 +88,18 @@ ProtoFile.prototype.register = function (item) {
   if (item instanceof Extend) {
     return; // TODO(gregp): handle extend
   }
-  if (item instanceof Message) {
+  if (item instanceof Message || item instanceof Enum) {
```
Comment (ninowalker) [inline parser/protofile.js]: I'll split these.

**Comment by ninowalker** [comment]
Currently errors:

```
  pbcljs
    ✓ is
    ✓ constructs & checks templates (709ms)
    ✓ gets expected ProtoFileSet (285ms)
    ✓ checks directories and cleans
    ✓ builds expected task map
    ◦ writes the expected files: (node:52007) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
(node:52007) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
    1) writes the expected files
    ◦ writes what we want: (node:52007) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
(node:52007) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.
    2) writes what we want


  ✖ 2 of 37 tests failed:

  1) pbcljs writes the expected files:
     Error: timeout of 10000ms exceeded
      at Timeout.<anonymous> (/Users/nino/dev/pbjs/node_modules/mocha/lib/runnable.js:167:14)
      at ontimeout (timers.js:488:11)
      at tryOnTimeout (timers.js:323:5)
      at Timer.listOnTimeout (timers.js:283:5)

  2) pbcljs writes what we want:
     Error: timeout of 10000ms exceeded
      at Timeout.<anonymous> (/Users/nino/dev/pbjs/node_modules/mocha/lib/runnable.js:167:14)
      at ontimeout (timers.js:488:11)
      at tryOnTimeout (timers.js:323:5)
      at Timer.listOnTimeout (timers.js:283:5)
```

**Comment by ninowalker** [comment]
This error is related to the modern node version; master has the same symptoms.

**Comment by ross-pfahler** [comment]
+1

**Pull request merged**